### ISSUE TITLE
JobUpdater failing with CouchDB internal_server_error

### ISSUE THREAD

**Issue by amaltaro** [dmwm/WMCore#11240]
**Impact of the bug**
WMAgent / CouchDB

**Describe the bug**
Every few days one of the components go down because they hit an exception when querying local CouchDB, the error is:
```
Message: CouchInternalServerError - reason: Internal Server Error, data: {'reduce': 'false', 'stale': 'update_after'} result: b'{"error":"internal_server_error","reason":"No DB shards could be opened.","ref":1577439538}\n'
```

and full traceback can be found at [1].

Standard 2.0.4.patch1 agent comes with the default `max_dbs_open` CouchDB parameter, but this agent had been tweaked to:
```
couchdb/max_dbs_open=500
```

and still it doesn't seem to be enough.

**How to reproduce it**
Apparently, by having too many tasks running in CouchDB (replication, indexing, compaction, etc).

**Expected behavior**
We should tweak the CouchDB configuration - and if needed the unix host limits - such that we don't hit this error.

This GH issues suggests two possible configuration improvements for CouchDB:
https://github.com/apache/couchdb/issues/796
by adding:
```
couchdb/max_dbs_open = ??? (a large value)
couchdb/update_lru_on_read = false
```

We don't have more than 10 databases, so I find it hard to believe that the current setting of max_dbs to 500 is not enough.

**Additional context and error message**
[1]
```
2022-08-02 21:49:01,870:139739194451712:INFO:JobUpdaterPoller:Synchronizing priorities with ReqMgr...
2022-08-02 21:49:10,217:139739194451712:ERROR:BaseWorkerThread:Error in worker algorithm (1):
Backtrace:
  <WMComponent.JobUpdater.JobUpdaterPoller.JobUpdaterPoller object at 0x7f1799bb89d0> <@========== WMException Start ==========@>
Exception Class: JobUpdaterException
Message: CouchInternalServerError - reason: Internal Server Error, data: {'reduce': 'false', 'stale': 'update_after'} result: b'{"error":"internal_server_error","reason":"No DB shards could be opened.","ref":1577439538}\n'
        ClassName : None
        ModuleName : WMComponent.JobUpdater.JobUpdaterPoller
        MethodName : algorithm
        ClassInstance : None
        FileName : /data/srv/wmagent/v2.0.4.patch1/sw/slc7_amd64_gcc630/cms/wmagentpy3/2.0.4.patch1/lib/python3.8/site-packages/WMComponent/JobUpdater/JobUpdaterPoller.py
        LineNumber : 102
        ErrorNr : 0

Traceback: 
  File "/data/srv/wmagent/v2.0.4.patch1/sw/slc7_amd64_gcc630/cms/wmagentpy3/2.0.4.patch1/lib/python3.8/site-packages/WMComponent/JobUpdater/JobUpdaterPoller.py", line 84, in algorithm
    self.synchronizeJobPriority()

  File "/data/srv/wmagent/v2.0.4.patch1/sw/slc7_amd64_gcc630/cms/wmagentpy3/2.0.4.patch1/lib/python3.8/site-packages/WMComponent/JobUpdater/JobUpdaterPoller.py", line 141, in synchronizeJobPriority
    workflowsToCheck = self.workqueue.getAvailableWorkflows()

  File "/data/srv/wmagent/v2.0.4.patch1/sw/slc7_amd64_gcc630/cms/wmagentpy3/2.0.4.patch1/lib/python3.8/site-packages/WMCore/Services/WorkQueue/WorkQueue.py", line 212, in getAvailableWorkflows
    data = self.db.loadView('WorkQueue', 'elementsDetailByWorkflowAndStatus',

  File "/data/srv/wmagent/v2.0.4.patch1/sw/slc7_amd64_gcc630/cms/wmagentpy3/2.0.4.patch1/lib/python3.8/site-packages/WMCore/Database/CMSCouch.py", line 571, in loadView
    retval = self.get('/%s/_design/%s/_view/%s' % \

  File "/data/srv/wmagent/v2.0.4.patch1/sw/slc7_amd64_gcc630/cms/wmagentpy3/2.0.4.patch1/lib/python3.8/site-packages/WMCore/Services/Requests.py", line 134, in get
    return self.makeRequest(uri, data, 'GET', incoming_headers,

  File "/data/srv/wmagent/v2.0.4.patch1/sw/slc7_amd64_gcc630/cms/wmagentpy3/2.0.4.patch1/lib/python3.8/site-packages/WMCore/Database/CMSCouch.py", line 135, in makeRequest
    self.checkForCouchError(getattr(e, "status", None),

  File "/data/srv/wmagent/v2.0.4.patch1/sw/slc7_amd64_gcc630/cms/wmagentpy3/2.0.4.patch1/lib/python3.8/site-packages/WMCore/Database/CMSCouch.py", line 171, in checkForCouchError
    raise CouchInternalServerError(reason, data, result)
```

**Comment by amaltaro** [comment]
I forgot to mention that this happens on CouchDB 3.1.2, maybe it will behave better once we upgrade it to CouchDB 3.2.2...

**Comment by amaltaro** [comment]
Ok, I modified submit4 CouchDB configuration with the following:
```
[couchdb]
max_dbs_open = 500
update_lru_on_read = false
```

and restarted components and couchdb. 
While restarting CouchDB, I noticed that not all of the processes were killed, there were a bunch of `couchjs` processes still running. I couldn't see what the problem is, the `ps` command in the manage script should be covering it as well. Anyways, I killed all those processes and restart everything. So far so good, but let's see in the coming days.
